getting-started-with-css (#2312)
section-engineering-education/engineering-education: content/articles/getting-started-with-css/index.md

# GETTING STARTED WITH CSS

> CSS is an abbreviation for cascading style sheets. It is a  language used to style the display of a document  written in a markup language like HTML. Alongside HTML and JavaScript, CSS is a foundation technology of the worldwide web.

Reviewer: ```suggestion
CSS is an abbreviation for cascading style sheets. It is a  language used to style the display of a document  written in a markup language like HTML. Alongside HTML and JavaScript, CSS is a foundational technology of the worldwide web.
```

## Functions of CSS
CSS helps developers change the presentation of web pages and make them responsive to different types of gadgets. It also helps them apply elements like lists and headings consistently throughout the site pages.

In this article I will explain how to select elements using different styles,doing various manipulations to the elements selected, changing the font and width of an element and lastly, using CSS to style HTML.
## 1.Selecting Elements
### Using the Elements' Name

> Developers can select the element uing the element's tag for example `p`,`h2`,`h1` etc as I have shown in the example below.

Reviewer: ```suggestion
Developers can select an element using its tag for example `p`,`h2`,`h1` etc as I have shown in the example below.
```

```html
<style>
   p{}
</style>
```

> `p` is the element's name which stands for paragraph.

Reviewer: ```suggestion
`p` in this case is the selector. It targets all paragraphs in the HTML document.
```

### Using a Class Name

> A class declaration is done by putting a fullstop before the class name in a style block for example `.red-text`then followed by a declaration block`{}`. Using a class name to select an element will be as shown below.

Reviewer: ```suggestion
To use an element's class as a selector, you need to put a full-stop before the class name in a style block for example `.red-text` followed by a declaration block`{}` as shown below.
```

```html
<style>
   .red-text{}
</style>
```

### Using an Id Selector

> An element can be selected with a specific Id by writing a `#` character followed by the element's Id. Id attributes are only applicable to one element. An Id attribute has a higher dominance than a class declaration, therefore, prioritized if they collide.An example of an Id selector is like`#Properties-of-CSS`. It can be set in HTML like this;

Reviewer: ```suggestion
To use an elements id as a selector you prepend it with the `#` character followed by the element's Id. Id attributes are only applicable to one element. An Id attribute is more specific when compared to a class declaration, therefore, prioritized if they collide. You can read [more about specificity on CSS Tricks](https://css-tricks.com/specifics-on-css-specificity/). An example of an Id selector is like `#Properties-of-CSS`. It can be set in HTML like this;
```

```html
<h2 id= "Properties-of-CSS">
 ```

>  Using CSS can style Id attributes like this;

Reviewer: ```suggestion
You can then select an element by id like so:
```

 ```html
 <style>
  #Properties-of-CSS{}
 </style>
 ```

### Using Inline Styles

> Developers set CSS to a specified HTML element by using the `style` attribute for example;

Reviewer: ```suggestion
Although discouraged, you can add styling to elements using inline styles. In the example below we set the color of the `h2` element to red using inline styles. We achieve this by using the `style` attribute.
```

```html
<h2 style=" color: red;">Properties of CSS </h2>
```

> ## 2.Doing Various Manipulations to the Elements Selected

Reviewer: ```suggestion
## 2. Different Ways to Manipulate Colors in CSS
```

### Using Names of Colors
With CSS you can change the text. To do this you need to target the element who's text you need to modify with the appropriate selector. You then use the `color`attribute to specify the color you desire for your text as shown below. This can either be done in a separete style sheet or in your HTML file by adding the `style`element like I have in the example.
```html
<style>
 h2{
     color: red;
 }
 </style>
 ```
 This will change all the h2 elements to the color red.
 

>  Although not advisable in practise,you can also use inline styles to achieve the same results as illustrated in the example below
>  ```html
>  <h2 style="color: red;">Properties of CSS</h2>
>  ```
>  As pointed out above, a class is declared by adding a fullstop(.) accompanied by the class name in a style block.Then followed by a declaration block({}).Using the name of a color to style the class declaration will be like this;
>  ```html
>  <style>
>    .red-text{
>        color: red;
>    }
>  </style>
>  ```
>  In this example,.red-text is the class declaration.
> 
> Introducing an Id attribute of Properties-of-CSS and giving it  a background color of black will be as shown in the example below.
> ```html
> <style>
>  #Properties-of-CSS{
>     background-color: black;
>  }
> </style>
> ```

Reviewer: This section is a repetition of the sections above. You had already gone over the different ways of targeting elements. You don't need to revisit that. Consider removing it entirely or finding a way to make it add value to the article.

```suggestion
```

### Using Hex Codes

> Hex is the short name for Hexadecimal code. Hex code uses six hexadecimal numbers to specify colors in CSS. Some Hex values of different colors according to [Simmons](http://web.simmons.edu/) are;

Reviewer: The Simmons link does not go to a resource that aligns with your citation. Update the link to point to the appropriate resource or consider removing the citation all together.

- black  #000000

- maroon  #800000

- yellow  #FFFF00

- purple  #800080

> Representing color in Hex code will be like;

Reviewer: ```suggestion
How to represent a color in Hex code is shown below:
```

```html
<style>
   p{
   color: #800080;
   }
</style>
```
 But since it is hard to remember hex code,one can shorten it. This are examples from [freshersnow](https://tutorials.freshersnow.com/css/css-colors/);
 - black  #000

 - red  #F00

 - cyan  #0FF

### Using RGB colors

> RGB stands for red, green and blue. It is another way of representing colors in CSS.Each value could have a value from between o to 100% or 0 to 255. A value of 0 means none of the colors was used while a value of 255 or 100% means all of that color was used. The color black would result from all values (red, green and blue)being 0 while the color white would result from all the values being a 100%. Some of RGB values according to [rapidtables](https://www.rapidtables.com/web/color/RGB_Color.html) are as follows;

Reviewer: ```suggestion
RGB stands for red, green and blue. It is another way of representing colors in CSS. Each value could have a value from between 0 to 255. A value of 0 means none of the colors was used while a value of 255 means all of that color was used. The color black would result from all values (red, green and blue) being 0 while the color white would result from all the values being a 255.

Here are some RGB values according to [rapidtables](https://www.rapidtables.com/web/color/RGB_Color.html):
```

- black  rgb(0, 0, 0)

- red  rgb(255, 0, 0)

- grey  rgb(128, 128, 128)

> RGB values can be styled in CSS as follows;

Reviewer: ```suggestion
How to represent RGB values in CSS is shown below:
```

```html
<style>
   p{
   color: rgb(128, 128, 128);
   }
</style>
```

> Let us use RGB values to represent an Id attribute;
> ```html
> <style>
>    #Properties-of-CSS{
>                 color: rgb(128, 128, 128);
>                 }
> </style>
> ```
> 
> 
> 

Reviewer: This is repetitive as you have already shown how to select elements in different ways exhaustively above. 
```suggestion
```

## 3.Changing the Font Size and Font Family of an Element
In order to change the font size of an element,the **font-size** element is introduced like this;
```html
h2{
    font-size: 15px;
}
```
Say for example one wants to change the font-size of a paragraph to 17px within the style element it will be like;

 ```html
p{
    font-size: 17px;
}
</style>
```

> In order to change the font name of an element,the **font-name** element is introduced like this;

Reviewer: You should use the appropriate words to refer to things eg. "element vs property ".
Using the wrong wording will mislead the reader.

```suggestion
In order to change the font name of an element,the **font-family** property is introduced like this;
```

```html
<style>
 p{
     font-family: Helvetica;
 }
</style>
```
## 4.Sizing Images

> CSS uses a property known as **width** to size elements by the help of class declarations such as **.smaller-width** and **.larger-image** like this;

Reviewer: Having the class declarations tied to this statement implies that they work together to achieve changes to an elements width. This is not the case. Consider removing that relation. 

```suggestion
CSS uses a property known as **width** to manipulate the width of elements. The use of the width property is shown below:
```

```html
<style>
 .smaller-image{
     width: 120px;
 }
 </style>
```
This should also be incorporated into your HTML element like this;
```html
<img class= "smaller-image">
```

## 5.Styling HTML Using CSS
  Let us create a HTML element and style it using CSS;
  ```html
  <!DOCTYPE html>
  <html>
  <h1>Getting Started With CSS</h1>
  <style>
   body{
       color: black;
       background-color: blue;
       font-family: monospace;
       font-size: 16px;
   }
   </style>
   </html>

>    ```

Reviewer: This section should tie together everything you've shown the reader. It does not do that. Consider revisiting the section and adding content that would tie together the article in a manner that adds value to it.

>    These are just fe guidelines to get you started with CSS but there is a lot more to learn. You can learn it from here [CSS](https://webplatform.github.io/docs/css)

Reviewer: ```suggestion
These are just a few guidelines to get you started with CSS but there is a lot more to learn. A good place to start would be the [web platform CSS docs](https://webplatform.github.io/docs/css)
```
